components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/internal/IdentityUserProfileServiceComponent.java

```diff
@@ -73,6 +75,16 @@ protected void activate(ComponentContext ctxt) {
             } else {
                 log.error("User profile management - ProfileMgtEventListener could not be registered.");
             }
+            AbstractAdmin userProfileAdmin = new UserProfileAdmin();
+            ServiceRegistration profileAdminSR = ctxt.getBundleContext().registerService(AbstractAdmin.class.getName(),
```

Comment:
Why are we adding a Admin service class as a osgi service? Ideally you should have a osgi service and the osgi service should be consumed by the admin service.